File: Leetcode 0690 Employee Importance [easy] [tree]/Employee Importance [bfs].js
/**
 * Definition for Employee.
 * function Employee(id, importance, subordinates) {
 *     this.id = id;
 *     this.importance = importance;
 *     this.subordinates = subordinates;
 * }
 */

/**
 * @param {Employee[]} employees
 * @param {number} id
 * @return {number}
 */
var GetImportance = function(employees, id) {
    let importance = 0;
    let idEmpMapping = new Map();
    for (let emp of employees) {
        idEmpMapping.set(emp.id, emp);
    }
    
    let queue = [];
    queue.push(idEmpMapping.get(id));
    while (queue.length > 0) {
        let size = queue.length;
        console.log(size);
        for (let i = 0; i < size; i++) {
            let cur = queue.shift();
            importance += cur.importance;
            for (let childId of cur.subordinates) {
                queue.push(idEmpMapping.get(childId));
            }
        }
    }
    
    return importance;
};

File: Leetcode 0690 Employee Importance [easy] [tree]/Employee Importance [bfs].java
/*
// Definition for Employee.
class Employee {
    public int id;
    public int importance;
    public List<Integer> subordinates;
};
*/

class Solution {
    public int getImportance(List<Employee> employees, int id) {
        int importance = 0;
        Map<Integer, Employee> idEmpMapping = new HashMap<>();
        for (Employee emp : employees) {
            idEmpMapping.put(emp.id, emp);
        }
        
        Queue<Employee> queue = new LinkedList<>();
        queue.offer(idEmpMapping.get(id));
        while (!queue.isEmpty()) {
            int size = queue.size();
            for (int i = 0; i < size; i++) {
                Employee cur = queue.poll();
                importance += cur.importance;
                for (int childId : cur.subordinates) {
                    queue.offer(idEmpMapping.get(childId));
                }
            }
        }
        
        return importance;
    }
}